core/src/main/java/io/questdb/griffin/engine/functions/eq/EqDoubleFunctionFactory.java

```diff
@@ -109,7 +109,7 @@ public Func(Function left, Function right) {
         public boolean getBool(Record rec) {
             final double l = left.getDouble(rec);
             final double r = right.getDouble(rec);
-            return negated != (l != l && r != r || Math.abs(l - r) < 0.0000000001);
```

Comment:
perhaps this would deserve a comment. it's not immediately obvious why `Math.abs(l - r) < 0.0000000001` is insufficient.

Reply:
OK, I'll add it now!